src/workerd/api/actor.c++

```diff
@@ -172,4 +173,32 @@ jsg::Ref<DurableObjectNamespace> DurableObjectNamespace::jurisdiction(kj::String
       idFactory->cloneWithJurisdiction(jurisdiction));
 }
 
+kj::Promise<void> DurableObjectNamespace::destroy(jsg::Lock& js, jsg::Ref<DurableObjectId> id) {
+  // We don't care about creating an actor just to destroy it.
+  // But for now let's keep it because workerd only supports GET_OR_CREATE mode for getting actor stubs.
+  // TODO(now): disable GET_OR_CREATE for destroy.
+  return destroyImpl(js, kj::mv(id), ActorGetMode::GET_OR_CREATE);
+}
+
+kj::Promise<void> DurableObjectNamespace::destroyExisting(jsg::Lock& js, jsg::Ref<DurableObjectId> id) {\
+  return destroyImpl(js, kj::mv(id), ActorGetMode::GET_EXISTING);
+}
+
+kj::Promise<void> DurableObjectNamespace::destroyImpl(jsg::Lock& js, jsg::Ref<DurableObjectId> id, ActorGetMode mode) {
+  JSG_REQUIRE(idFactory->matchesJurisdiction(id->getInner()), TypeError,
```

Comment:
Non-blocking nit: this really isn't a `TypeError`. A regular `Error` would be more appropriate here.